doc/main/reference/blocked_nd_range_ctad.rst

```diff
@@ -0,0 +1,161 @@
+    oneapi::tbb::blocked_range<int> range2(0, 200);
+    oneapi::tbb::blocked_range<int> range3(0, 300);
+
+    // Since 3 unidimensional ranges of type int are provided, the type of nd_range
+    // can be deduced as oneapi::tbb::blocked_nd_range<int, 3>
+    oneapi::tbb::blocked_nd_range nd_range(range1, range2, range3);
+
+API
+***
+
+.. code:: cpp
+    
+    #include <oneapi/tbb/blocked_nd_range.h>
+
+Synopsis
+--------
+
+.. code:: cpp
+
+    namespace oneapi {
+    namespace tbb {
+
+        template <typename Value, unsigned int N>
+        class blocked_nd_range {
+        public:
+            // Member types and constructors defined as part of oneTBB specification
+            using value_type = Value;
+            using dim_range_type = blocked_range<value_type>;
+            using size_type = typename dim_range_type::size_type;
+
+            blocked_nd_range(const dim_range_type& dim0, /*exactly N arguments of type const dim_range_type&*/); // [1]
+            blocked_nd_range(const value_type (&dim_size)[N], size_type grainsize = 1);                          // [2]
+            blocked_nd_range(blocked_nd_range& r, split);                                                        // [3]
+            blocked_nd_range(blocked_nd_range& r, proportional_split);                                           // [4]
+        }; // class blocked_nd_range
+
+        // Explicit deduction guides
+        template <typename Value, typename... Values>
+        blocked_nd_range(blocked_range<Value>, blocked_range<Values>...)
+        -> blocked_nd_range<Value, 1 + sizeof...(Values)>;
+
+        template <typename Value, unsigned int... Ns>
+        blocked_nd_range(const Value (&...)[Ns])
+        -> blocked_nd_range<Value, sizeof...(Ns)>;
+
+        template <typename Value, unsigned int N>
+        blocked_nd_range(const Value (&)[N], typename blocked_nd_range<Value, N>::size_type = 1)
+        -> blocked_nd_range<Value, N>;
+
+        template <typename Value, unsigned int N>
+        blocked_nd_range(blocked_nd_range<Value, N>, split)
+        -> blocked_nd_range<Value, N>;
+
+        template <typename Value, unsigned int N>
+        blocked_nd_range(blocked_nd_range<Value, N>, proportional_split)
+        -> blocked_nd_range<Value, N>;
+    } // namespace tbb
+    } // namespace oneapi
+
+Deduction Guides
+----------------
+
+Copy and move constructors of ``blocked_nd_range`` provide implicitly-generated deduction guides. 
+In addition, the following explicit are provided:
+
+.. code:: cpp
+
+    template <typename Value, typename... Values>
+    blocked_nd_range(blocked_range<Value>, blocked_range<Values>...)
+    -> blocked_nd_range<Value, 1 + sizeof...(Values)>;
+
+**Effects**: Allows deduction from a set of ``blocked_range`` objects provided to the constructor ``[1]`` of ``blocked_nd_range``.
+**Constraints**: This deduction guide participates in overload resolution only if all of the types in `Values` are same as `Value`.
```

Comment:
```suggestion
**Constraints**: Participates in overload resolution only if all types in `Values` match `Value`.
```

Reply:
Partially applied - I prefer to keep "same as" at the end since "match" is not clear for me from the C++ perspective and "same as" matches the corresponding `std::is_same` API.